source/ScreensaverHero.hx

```diff
@@ -0,0 +1,92 @@
+package;
+
+import flixel.FlxG;
+import flixel.FlxSprite;
+import flixel.math.FlxRandom;
+import flixel.util.FlxSpriteUtil;
+
+class ScreensaverHero extends FlxSprite
+{
+	public var direction:Int;
+
+	public function new(x:Int, y:Int)
+	{
+		super(x, y);
+		var randomDirection:FlxRandom = new FlxRandom();
+		this.direction = randomDirection.int(0, 7);
+	}
+
+	override function update(elapsed:Float)
+	{
+		if (this.x < 500)
+		{
+			this.color = 0xff0000;
+		}
+		if (this.x > 500)
+		{
+			this.color = 0x00ff00;
+		}
+		if (this.direction == 0)
+		{
+			this.x--;
+		}
+		else if (this.direction == 1)
+		{
+			this.x--;
+			this.y--;
+		}
+		else if (this.direction == 2)
+		{
+			this.y--;
+		}
+		else if (this.direction == 3)
+		{
+			this.x++;
+		}
+		else if (this.direction == 4)
+		{
+			this.x++;
+			this.y++;
+		}
+		else if (this.direction == 5)
+		{
+			this.y++;
+		}
+		else if (this.direction == 6)
+		{
+			this.x--;
+			this.y++;
+		}
+		else
+		{
+			this.x++;
+			this.y--;
+		}
+		if (FlxG.mouse.pressed)
+		{
+			this.direction++;
+			if (this.direction == 8)
+			{
+				this.direction = 0;
+			}
```

Comment:
If you're sticking with this same direction plan, you can simplify this to 
```
direction = direction % 8;
```
since modulo will reset the value to zero if it's reached 8.